phpseclib/Net/SFTP.php

```diff
@@ -811,6 +811,17 @@ private function nlist_helper($dir, $recursive, $relativeDir)
         return $result;
     }
 
+    /**
+     * Yields information about files in the given directory.
+     *
+     * @param string $dir
+     * @return \Generator
+     */
+    public function nlistGenerator($dir = '.')
+    {
+        yield from $this->readlistGenerator($dir);
```

Comment:
An alternative for 5.5 should not be an issue if compatibility is required.

Reply:
To be included in the 3.0 branch 5.6 compatibility is a requirement but the master branch should always be assumed to have an unstable API and is not at all aiming to maintain BC with itself. Overall, I'm okay bumping the master branch version reqs to 7.0.

I'll try to review your PR more thoroughly this evening - thanks!